src/main/scala/mirroring/services/writer/CustomChangeTrackingService.scala

```diff
@@ -49,33 +49,44 @@ class CustomChangeTrackingService(
   val userMetadataJSON: String            = generateUserMetadataJSON(context.ctCurrentVersion)
 
   override def write(data: DataFrame): Unit = {
-    if (DeltaTable.isDeltaTable(this.getSparkSession, context.path)) {
+    if (!DeltaTable.isDeltaTable(getSparkSession, context.path)) {
+      logger.info("Target table doesn't exist yet. Initializing table...")
+      super.write(data)
+    } else {
       logger.info("Target table already exists. Merging data...")
-      verifySchemaMatch(data)
+      mergeData(data)
+    }
+  }
+
+  private def mergeData(data: DataFrame): Unit = {
+    verifySchemaMatch(data)
 
+    val updOperations = data.as(Config.SourceAlias).where(insertCondition)
```

Comment:
do we really need to make things so complicated? what if there are no delete operations? will it still read the entire table to make a commit?

Reply:
If there are no delete operation, `deleteRows` with executing merge leads to SnaphotIsolation commit without reading the whole table.

Reply:
ok then